provider/ethereum/token/client.go

```diff
@@ -243,10 +248,28 @@ func (c *client) lookupERC20ByRPC(ctx context.Context, chainID uint64, address c
 	return &tokenMetadata, nil
 }
 
+func (c *client) lookupERC20ByRedis(ctx context.Context, chainID uint64, address common.Address) (*metadata.Token, error) {
```

Comment:
all func names are very similar, so adding a line of comment for all will be useful.